### PR TITLE
Fix unicycle counting as car

### LINKED ISSUE #246: Unicycle is counted as a vehicle if you do not have a vehicle spawned
I have found that when in a session, if you spawn a unicycle and not a vehicle then when you go to spawn a vehicle after the server will not allow you.

If you remove your unicycle first and then spawn a vehicle it will allow you.

It looks as though the server counts your unicycle on some occasions as if you spawn the unicycle once you have a vehicle in a 1 vehicle per player server it will spawn as expected.

Does this make sense?

### PR DESCRIPTION
**Pull request by sla-ppy**
Fixes #246.

### CODE REVIEW

**Comment by sla-ppy** [comment]
This is a super dirty fix to the issue.
We essentially duct tape a problem which should definitely be revisited later on!

The reason for this dirty fix is because mVehichleData contains both unicycle and vehicle data, and I am unsure if we want it to stay like that.

I propose that in the new protocol we treat unicycle and vehicle data differently because there are questions to be answered like:
- Do we want to allow players to spawn multiple unicycles (as controllable props) or do we want to have a single one representing and acting as the player itself?
- Does it truly make sense to treat unicycles and vechicles differently? 

Benefit of the proposal:
- Counting the unicycle and vehicle counts separately could be toyed around with further.
- Future wise the unicycle could have multiple models for players to have fun with.

**Comment by sla-ppy** [comment]
Please review my code: @lionkor.

File: src/Client.cpp [hunk 1]
```diff
@@ -120,7 +120,15 @@ void TClient::SetCarData(int Ident, const std::string& Data) {
     DeleteCar(Ident);
 }
 
+// TODO: count unicycles and vehicles different #246    sla-ppy
+// PROPOSAL: separate unicycles completely from vehicles???
```
Comment (lionkor) [inline src/Client.cpp]: please remove these comments lines 123 and 124

File: src/Client.cpp [hunk 2]
```diff
@@ -120,7 +120,15 @@ void TClient::SetCarData(int Ident, const std::string& Data) {
...
 int TClient::GetCarCount() const {
+    std::unique_lock lock(mVehicleDataMutex);
+    for (auto& v : mVehicleData) {
```
Comment (lionkor) [inline src/Client.cpp:127]: Please put a short(ish) comment here explaining the *reasoning* for this solution: For example, it should outline what this fixes, and why this is the fix that was chosen ("it works lol").
You can then remove the comment on L129.

**Review by lionkor — CHANGES_REQUESTED** [review]
Thanks, logic looks sound and we obv saw it work.

**Comment by Starystars67** [comment]
Could I further this topic by suggesting that instead a configuration option be added to specify vehicle names that can be added so they bypass the vehicle limit.

This would be for things like the unicycle as default but can then be expanded to items like cones, bollards, concrete barriers or even pucks.

**Comment by lionkor** [comment]
> Could I further this topic by suggesting that instead a configuration option be added to specify vehicle names that can be added so they bypass the vehicle limit.
> 
> This would be for things like the unicycle as default but can then be expanded to items like cones, bollards, concrete barriers or even pucks.

I would ask you to make a new feature issue for that, because I'd also like to specify that properly before building it, and I wouldn't want to tack it onto this bug as I want this in the next release.

**Comment by lionkor** [comment]
@sla-ppy btw in `documentate unicycle dirty fix` the correct form is `document` ;)

**Review by lionkor — APPROVED** [review]
looks good

**Comment by sla-ppy** [comment]
> @sla-ppy btw in `documentate unicycle dirty fix` the correct form is `document` ;)

catty

**Pull request merged**